packages/models-library/src/models_library/utils/specs_substitution.py

```diff
@@ -55,22 +63,37 @@ def substitutions(self):
         return self._substitutions
 
     def set_substitutions(
-        self, environs: dict[str, SubstitutionValue]
+        self, mappings: dict[str, SubstitutionValue] | None = None
     ) -> SubstitutionsDict:
-        """NOTE: ONLY targets identifiers declared in the specs"""
-        identifiers_needed = self.get_identifiers()
-
+        """
+        NOTE: ONLY targets identifiers declared in the specs
+        NOTE:`${identifier:-a_default_value}` will replace the identifier with `a_default_value`
+        if not provided
+        """
+        if mappings is None:
+            mappings = {}
+
+        needed_identifiers = self.get_identifiers()
+
+        needed_identifiers_with_defaults: dict[str, str | None] = {}
+        for identifier in needed_identifiers:
+            parts = identifier.split(":-")
+            needed_identifiers_with_defaults[identifier] = (
+                parts[1] if ":-" in identifier else None
+            )
+
+        resolved_identifiers: dict[str, str] = {}
+        for identifier in needed_identifiers_with_defaults:
+            if identifier in mappings:
+                resolved_identifiers[identifier] = cast(str, mappings[identifier])
+            elif needed_identifiers_with_defaults[identifier] is not None:
```

Comment:
here will raise a `KeyError` if a `required_identifier` is undefined and has no default defined. IMO that error should be at least categorized here

Reply:
No it is not the case. The `identifier` key is already part of the `needed_identifiers_with_defaults` dict. Let me rewrite the code to avoid this misleading code.